packages/app/src/flame/valibot/schemaUtil.ts

```diff
@@ -0,0 +1,11 @@
+import { f32, struct } from 'typegpu/data'
+
+export function schemaToF32Struct<T extends Record<string, unknown>>(
+  schema: T,
+) {
+  const obj = {} as Record<keyof T, typeof f32>
+  for (const key of Object.keys(schema) as (keyof T)[]) {
```

Comment:
you can use the `recordKeys` utility so you don't have to cast

Reply:
True, fixed